### ISSUE TITLE
Reorganise test-runners

### ISSUE THREAD

**Issue by pozorvlak** [Growstuff/growstuff#1125]
What I'd like is something along the following lines:

- `rake js_spec` runs all JavaScript unit tests
- `rake ruby_spec` runs all Ruby unit tests
- `rake spec` runs all unit tests
- `rake static` runs all static checks (rubocop, contributors.md, Gemfile, eventually haml-lint, etc)
- `rake` runs all the above
- we run three Travis jobs, as follows:
  - `GROWSTUFF_ELASTICSEARCH=true rake spec`
  - `GROWSTUFF_ELASTICSEARCH=false rake spec`
  - `rake static`

So running `rake` locally should do a pretty good job of testing everything we're going to test in CI (modulo ElasticSearch), which should make it easier to not get caught out by avoidable failures.

I think this should all be pretty straightforward to implement - what do you all think?

**Comment by maco** [comment]
Sounds good to me

**Comment by CloCkWeRX** [comment]
So we already have
```
rake spec:controllers                                    # Run the code exa...
rake spec:features                                       # Run the code exa...
rake spec:helpers                                        # Run the code exa...
rake spec:lib                                            # Run the code exa...
rake spec:mailers                                        # Run the code exa...
rake spec:models                                         # Run the code exa...
rake spec:requests                                       # Run the code exa...
rake spec:routing                                        # Run the code exa...
rake spec:views  
```

I would be keen to 
 1. Static checks, etc 
 2. just run all of the js, ruby specs *minus* features first, and without elasticsearch; (ie; actual 'unit tests')
 3. then run features with elasticsearch on (more like end to end/integration tests)

As we move more towards automated deployment, if we beef up the end-to-end testing capability (though it'll be slower), it might be worth treating the unit tests as 'ok to merge', while the feature tests become more 'ok to deploy'

**Comment by Br3nda** [comment]
Fixed in #1133
